plugins/package-managers/bazel/src/main/kotlin/Bazel.kt

```diff
@@ -462,6 +523,116 @@ class Bazel(override val descriptor: PluginDescriptor = BazelFactory.descriptor)
 
             is ArchiveSourceInfo -> null
         }
+
+    /**
+     * Process the Conan dependencies [conanAnalyzerResults] returned by `resolveConanDependencies`, based on the
+     * [extension] information returned by 'bazel mod graph'. The newly created scopes are added to [scopes] and the
+     * Conan packages are returned.
+     */
+    private fun processConanDependencies(
+        extension: BazelExtension,
+        scopes: MutableSet<Scope>,
+        conanAnalyzerResults: List<ProjectAnalyzerResult>
+    ): MutableSet<Package> {
+        val conanPackages = mutableSetOf<Package>()
+        val conanProjectReferences = mutableSetOf<PackageReference>()
+        val conanProjectWithDependencies = mutableSetOf<PackageReference>()
+        val conanProjectWithTestDependencies = mutableSetOf<PackageReference>()
+
+        conanPackages += conanAnalyzerResults.map { it.project.toPackage() }
+
+        conanAnalyzerResults.forEach { conanAnalyzerResult ->
+            val projectReference = conanAnalyzerResult.project.toPackage().toReference()
+            conanProjectReferences += projectReference
+
+            val dependenciesPerScopes = conanAnalyzerResult.project.scopes.associate {
+                // Only include the packages reported by Bazel.
+                it.name to it.dependencies.filterTo(mutableSetOf()) { dep -> dep.id.name in extension.usedRepos }
+            }
+
+            conanProjectWithDependencies += projectReference.copy(
+                dependencies = dependenciesPerScopes["requires"].orEmpty()
+            )
+            conanProjectWithTestDependencies += projectReference.copy(
+                dependencies = dependenciesPerScopes["test_requires"].orEmpty()
+            )
+            conanPackages += conanAnalyzerResult.packages.filter { pkg ->
+                (conanProjectWithDependencies + conanProjectWithTestDependencies).flatMap {
+                    it.dependencies
+                }.any { it.id == pkg.id }
+            }
+        }
+
+        scopes.mapTo(mutableSetOf()) {
+            if (it.name == "main") {
+                it.copy(dependencies = it.dependencies + conanProjectReferences)
+            } else {
+                it
+            }
+        }.apply {
+            scopes.clear()
+            scopes.addAll(this)
+        }
```

Comment:
Nit: As you're mostly referring to `scopes` here, not `this`, using `apply` seems a bit odd. I'd propose

```kotlin
.also {
    scopes.clear()
    scopes.addAll(it)
}
```

instead.